hydromt_fiat/workflows/exposure_vector.py

```diff
@@ -598,9 +598,18 @@ def setup_ground_floor_height(
                     gfh = gfh.drop(columns=columns_to_drop)
         
                 gdf = self.get_full_gdf(self.exposure_db)
-                gdf = join_spatial_data(
-                    gdf, gfh, attribute_name, gfh_method, max_dist, self.logger
-                )
+                
+                # If roads in model filter out for spatial joint 
+                if gdf["Primary Object Type"].str.contains("roads").any():
```

Comment:
this change and the one at line 697 look really similar. Maybe make a function for that?